packages/amplify-category-plugin-interface/src/category-override-base.ts

```diff
@@ -0,0 +1,16 @@
+import { $TSContext, $TSAny } from 'amplify-cli-core';
+import { Template } from './amplifyBaseCDKTypes';
+
+export abstract class AmplifyCategoryTransform {
```

Comment:
the essential functions in here which should be implmeneted by all classes are:

transform
applyOverrides
saveBuildFiles

rest of them are just helper functions for the transform() functions.